File: 1768 - Merge Strings Alternately/PHP/Solution.php
<?php

class Solution {

/**
 * @param String $word1
 * @param String $word2
 * @return String
 */
function mergeAlternately($word1, $word2) {
    $res = "";
    $max = max(strlen($word1), strlen($word2));
    
    for ($i = 0; $i < $max; $i++) {
        if ($i < strlen($word1)) {
            $res .= $word1[$i];
        }
        
        if ($i < strlen($word2)) {
            $res .= $word2[$i];
        }
    }
    
    return $res;
}
}

File: 1768 - Merge Strings Alternately/Go/Solution.go
func mergeAlternately(word1 string, word2 string) string {
	res := ""
    max := int(math.Max(float64(len(word1)), float64(len(word2))))

	for i := 0; i < max; i++ {
		if i < len(word1) {
			res += string(word1[i])
		}
		if i < len(word2) {
			res += string(word2[i])
		}
	}

	return res
}
